### PR TITLE
Drop unsupported language switching commands for OneCore

### LINKED ISSUE #13732: Error with OneCore voice and automatic language switching in NVDA2022.1
### Context
Issue found while trying to reproduce #13727. Opened as a new issue as per https://github.com/nvaccess/nvda/issues/13727#issuecomment-1135409560.
Cc @seanbudd 

### Steps to reproduce:
* Use OneCore non-English voice (mine is Hortense - French)
* Ensure NVDA has automatic language switching enabled (during my tests, automatic dialect switching is disabled however)
* Open the following URL: `data:text/html,<p>Hello</p><p lang="en">world</p>`
* Read the page

### Actual behavior:
* With NVDA 2021.3.4 (portable), the whole page is read in French since I have no English OneCore voice.
* With NVDA 2022.1 (installed), the word "world" is not read and an error is logged.

Here are the logs for both versions of NVDA:
* [OneCoreWithNVDA2021.3.4.log](https://github.com/nvaccess/nvda/files/8755212/OneCoreWithNVDA2021.3.4.log): OK
* [OneCoreWithNVDA2022.1.log](https://github.com/nvaccess/nvda/files/8755214/OneCoreWithNVDA2022.1.log): Error

### Expected behavior:

OneCore reads the whole page; if no English voice is available the default (French) voice should be used instead.

### !additional note
If reading 5 times the line "world" causing the error, the synth switches to eSpeak. Cf log:
[OneCoreSwitchToESpeak.log](https://github.com/nvaccess/nvda/files/8763530/OneCoreSwitchToESpeak.log)

#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2022.1 
#### Windows version:
Windows 10 2004 (x64) build 19041.388
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
* Windows French, NVDA language default (thus French)
* In the list of OneCore voices, I have French voices (Hortense, Julie, Paul) and German voices (Stefan, Katja, Hedda) but no English voice. I do not remember if I had never had one, if I have uninstalled it (what is very unlikely but why not for tests), or if this voice has disappeared without my willing (e.g. during an update or an installation of something).

#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
As written earlier, 2021.3.4 (portable) does not exhibit this issue.
#### If NVDA add-ons are disabled, is your problem still occurring?
Test done without add-ons
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Fixes #13732 

### Summary of the issue:
OneCore keeps a cache of voices for NVDA in the registry.
OneCore fails to automatic language switch when remnants of a previously installed OneCore language exist in the cache.
This causes speech to not be announced.

This is only an issue for installed copies, as portable copies bypass the NVDA OneCore cache.

### Description of how this pull request fixes the issue:
Drop the language change command if a given language is unsupported by OneCore.

#### Manual testing

Using the following HTML sample, test language switching. 
Replace the language code "zh" as appropriate.

`data:text/html,<p>Hello</p><p lang="zh">world</p>`.

Using installed oneCore languages `'en_au', 'fr_fr', 'en_us', 'zh_hk'`

Automatic language switching is successful for the following language codes:
- "en" (to en_US)
- "en_AU"
- "en-AU"
- "en-au"
- "en_AU"
- "fr-fr"
- "fr" (to fr_FR)
- "zh" (to zh_hk)
- "zh_hk"

Automatic language switching reverts to the default voice for the following language codes:
- "FakeLang_fakeLocale" (illegitimate language)
- "en-CA" (unsupported locale)
- "ta" (unsupported language)

### Known issues with pull request:
None

### Change log entries:
Bug fixes
```t2t
- Fix automatic language switching failing with OneCore when trying to switch to a formerly installed language. (#13732)
```

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

File: source/synthDrivers/oneCore.py [hunk 1]
```diff
@@ -135,6 +163,9 @@ class OneCoreSynthDriver(SynthDriver):
 	}
 	supportedNotifications = {synthIndexReached, synthDoneSpeaking}
 
+	_ocSpeechToken: Optional[ctypes.POINTER]
+	_queuedSpeech: List[Union[str, Tuple[Callable[[ctypes.POINTER, float], None], float]]]
```
Comment (feerrenrut) [inline source/synthDrivers/oneCore.py:167]: Shouldn't this typing be in the `init` method? Doesn't having it here indicate these are expected to be used as class variables?

**Review by feerrenrut — CHANGES_REQUESTED** [review]
Overall looks good.

File: source/synthDrivers/oneCore.py [hunk 2]
```diff
@@ -74,23 +85,40 @@ def convertCharacterModeCommand(self, command):
...
 
-	def __init__(self, defaultLanguage, rate, pitch, volume):
-		super(_OcPreAPI5SsmlConverter, self).__init__(defaultLanguage)
+	def __init__(
+			self,
+			defaultLanguage: str,
+			availableLanguages: Set[str],
+			rate: float,
+			pitch: float,
+			volume: float,
```
[the remark this answers is not in the document]
Comment (seanbudd) [reply source/synthDrivers/oneCore.py:130]: This is actually from 0-100, but supports decimal changes

File: source/synthDrivers/oneCore.py [hunk 3]
```diff
@@ -74,23 +85,40 @@ def convertCharacterModeCommand(self, command):
...
 		return super().convertLangChangeCommand(command)
 
 class _OcPreAPI5SsmlConverter(_OcSsmlConverter):
 
-	def __init__(self, defaultLanguage, rate, pitch, volume):
-		super(_OcPreAPI5SsmlConverter, self).__init__(defaultLanguage)
+	def __init__(
+			self,
+			defaultLanguage: str,
+			availableLanguages: Set[str],
```
[the remark this answers is not in the document]
Comment (seanbudd) [reply source/synthDrivers/oneCore.py:127]: These are both fully qualified OneCore language codes, e.g. "en_US".

File: source/synthDrivers/oneCore.py (hunk 1 above)
Comment (seanbudd) [reply source/synthDrivers/oneCore.py]: Fixed in e630724

File: source/synthDrivers/oneCore.py [hunk 4]
```diff
@@ -7,7 +7,7 @@
 """
 
 import os
-import sys
+from typing import Any, Callable, Generator, List, Optional, Set, Tuple, Union
```
[the remark this answers is not in the document]
Comment (seanbudd) [reply source/synthDrivers/oneCore.py]: Fixed in e630724

**Comment by CyrilleB79** [comment]
Hello
I have just tested nvda_snapshot_pr13779-25537,4aba6cc9.exe:
* From the launcher, the test is OK. I get a warning for each English text, what is normal due to my broken English voices.
* From a portable copy, I get a critical error: [nvda-old.log](https://github.com/nvaccess/nvda/files/8873347/nvda-old.log)
  However, the issue is also present on last alpha portable (nvda_snapshot_alpha-25538,f3445949.exe): cf. [nvda-old.log](https://github.com/nvaccess/nvda/files/8873394/nvda-old.log)

**Comment by seanbudd** [comment]
@CyrilleB79 - it looks like you've found a new regression in 2022.2 for portable copies of NVDA.
This needs a new issue and investigation.

**Review by feerrenrut — APPROVED** [review]

**Review by feerrenrut — APPROVED** [review]
Assuming the build passes.

**Pull request merged**